toolchain/lower/file_context.cpp

```diff
@@ -50,7 +51,17 @@ auto FileContext::Run() -> std::unique_ptr<llvm::Module> {
     functions_[i] = BuildFunctionDecl(SemIR::FunctionId(i));
   }
 
-  // TODO: Lower global variable declarations.
+  // Lower global variable declarations.
+  for (auto inst_id :
+       sem_ir().inst_blocks().Get(sem_ir().top_inst_block_id())) {
+    // Only `VarStorage` indicates global variable declaration in the
+    // top instruction block.
+    if (auto inst = sem_ir().insts().Get(inst_id);
+        inst.kind() != SemIR::InstKind::VarStorage) {
+      continue;
+    }
+    global_variables_.Insert(inst_id, BuildGlobalVariableDecl(inst_id));
```

Comment:
What do you think of changing `BuildGlobalVariableDecl` to take a `SemIR::VarStorage` instead of an `inst_id`?

We have helpers like `insts().Is` and `insts().TryGetAs`, so you could write this like:

```suggestion
    if (auto var = sem_ir().insts().TryGetAs<SemIR::VarStorage>(inst_id)) {
      global_variables_.Insert(inst_id, BuildGlobalVariableDecl(*var));
    }
```

Note a subtle effect of that is it means the array lookup for `inst_id` only occurs once, since `BuildGlobalVariableDecl` directly reuses the `Get` result from here.

Reply:
Thanks! Done.